osbs/conf.py

```diff
@@ -266,12 +266,22 @@ def get_registry_secrets(self):
         else:
             return []
 
-    def get_registry_api_versions(self):
+    def get_registry_api_versions(self, platform=None):
         value = self._get_value("registry_api_versions",
                                 self.conf_section,
                                 "registry_api_versions",
                                 default='v1,v2')
-        return value.split(',')
+        versions = value.split(',')
+        section = 'platform:{0}'.format(platform)
```

Comment:
If `platform is None` we should not perform this check and instead return `versions` unchanged.

We should only remove v1 support if a platform is specified for which `enable_v1` is false.

This is because the orchestrator build, which is not platform-specific, needs to know the full set of API versions required. Only the worker builds, for which a platform will be specified, will need to know to disable v1 support.